### PR TITLE
Initial implementation of Hdf5RasPrecipDataWriter

### PR DESCRIPTION
**Pull request by tombrauer**

### CODE REVIEW

File: vortex-api/build.gradle.kts [hunk 1]
```diff
@@ -28,6 +28,9 @@ dependencies {
     implementation("systems.uom:systems-common:2.0.2")
     implementation("edu.ucar:cdm-core:5.5.3")
     implementation("org.apache.commons:commons-compress:1.20")
+    implementation("org.hdfgroup:hdfobject:2.14.0a")
+    implementation("org.hdfgroup:jarh5obj:2.10.1")
+    implementation("org.hdfgroup:jarhdf5:3.3.2")
```
Comment (tombrauer) [inline vortex-api/build.gradle.kts:33]: Are all 3 of these neccesary or just `hdfobject`? I recall you saying you used just the base lib.

File: vortex-api/src/main/java/mil/army/usace/hec/vortex/io/DataWriter.java [hunk 2]
```diff
@@ -91,6 +91,12 @@ public DataWriter build() {
                 return new NetcdfDataWriter(this);
             }
 
+            //PathMatcher hdf5Matcher = FileSystems.getDefault().getPathMatcher("glob:**/*.p**.{h5,hdf5}"); //trying to match a RAS plan output e.g. MyModel.p01.hdf
```
Comment (tombrauer) [inline vortex-api/src/main/java/mil/army/usace/hec/vortex/io/DataWriter.java:94]: Can you get the RAS "p" file path matching working? I think the hdf pattern matching will cause problems in the future if we ever implement a generalized HDF writer.

File: vortex-api/src/main/java/mil/army/usace/hec/vortex/io/Hdf5RasPrecipDataWriter.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        private int lastrow = 0;
+        private final String dataset_name;
+        private final String DefaultDatasetName = "Event Conditions/Meteorology/Precipitation/Values";
+
+        private final int max_rows = 999999;
+
+        public HdfRasPrecipDataset(List<VortexGrid> grids, Map<String,String> options) throws Exception{
+            this.grids=grids;
+            if (grids.size()==0){
+                throw new Exception("No grids in write operation"); //@TODO is this necessary?
```
Comment (tombrauer) [inline vortex-api/src/main/java/mil/army/usace/hec/vortex/io/Hdf5RasPrecipDataWriter.java:72]: if size == 0, I would just do nothing. You could add logged output.

**Review by tombrauer — COMMENTED** [review]
First draft review complete. Just a few comments.

File: vortex-api/src/main/java/mil/army/usace/hec/vortex/io/DataWriter.java (hunk 2 above)
Comment (rgoss) [reply vortex-api/src/main/java/mil/army/usace/hec/vortex/io/DataWriter.java:94]: fixed.

**Pull request merged**